backend/src/models/CommentManager.js

```diff
@@ -0,0 +1,102 @@
+const AbstractManager = require("./AbstractManager");
+
+class CommentManager extends AbstractManager {
+  constructor() {
+    // Call the constructor of the parent class (AbstractManager)
+    // and pass the table name "comment" as configuration
+    super({ table: "comment" });
+  }
+
+  // The C of CRUD - Create operation
+
+  async create(comment) {
+    // Execute the SQL INSERT query to add a new user to the "comment" table
+    const [result] = await this.database.query(
+      `INSERT INTO ${this.table} (comment_date_time, comment_content, user_id, decision_id) VALUES (?,?,?,?)`,
+      [
+        comment.comment_date_time,
+        comment.comment_content,
+        comment.user_id,
+        comment.decision_id,
+      ]
+    );
+    // Return the ID of the newly inserted comment
+    return result.insertId;
+  }
+
+  // The Rs of CRUD - Read operations
+
+  async read(id) {
+    // Execute the SQL SELECT query to retrieve a specific comment by its ID
+    const [rows] = await this.database.query(
+      `SELECT * FROM ${this.table} WHERE comment_id = ?`,
+      [id]
+    );
+
+    // Return the first row of the result, which represents the user
+    return rows[0];
+  }
+
+  // Execute the SQL SELECT query to retrieve a specific comment by the id of the decision to which it is associated
+  async readByDecision(id) {
+    const [rows] = await this.database.query(
+      `SELECT comment.comment_date_time, user.picture, CONCAT(user.firstname,' ', user.lastname) AS name, assignment.role, comment.comment_content
+          FROM ${this.table}
+          INNER JOIN assignment ON comment.user_id = assignment.user_id
+          INNER JOIN user ON comment.user_id = user.user_id
+          INNER JOIN decision ON comment.decision_id = decision.decision_id
+          WHERE decision.decision_id = ?
+          ORDER BY comment.comment_date_time DESC`,
+      [id]
+    );
+    return rows[0];
+  }
+
+  async readByRole(id) {
+    // Execute the SQL SELECT query to retrieve a specific comment's role as "expert", "impacted" or "visitor"
+    const [rows] = await this.database.query(
+      `SELECT role FROM assignment WHERE user_id = ?`,
+      [id]
+    );
+    return rows[0] ? "" : ""; /* ??? */
```

Comment:
why put a condition on it?

Reply:
it was a test, I forgot to delete it because it wasn't my ticket, hence the question marks in the comments right next to it.